logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -443,13 +443,19 @@ protected void onActorClosing() {
     metrics.close();
 
     if (!isFailed()) {
-      streamProcessor.onClose();
-      createSnapshot();
+      // since actor.run() makes the task the next executed job, we first schedule the state closing
```

Comment:
Maybe it is more intuitive if we use the actor lifecycle methods? Like onActorCloseRequested you do your streamProcessor.onClose and onClosing you do your close/teardown work or event do this in onActorClosed.